app/Http/Controllers/WorkflowController.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\EvidencioAPI;
+use App\Workflow;
+use App\LoadedEvidencioModel;
+use App\Step;
+use App\Field;
+use App\Option;
+use Illuminate\Support\Facades\Gate;
+use Illuminate\Validation\UnauthorizedException;
+
+/**
+ * WorkflowController class, handles database- and API-calls for Workflowpage.
+ */
+class WorkflowController extends Controller
+{
+
+    /**
+     * Show the application dashboard.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index($workflowId)
+    {
+        $result = WorkflowController::loadWorkflow($workflowId);
+        return view('workflow')->with('result', $result);
+    }
+
+    /**
+     * Loads a workflow from the database based on the workflowId
+     *
+     * @param Number $workflowId
+     * @return Array
+     */
+    public function loadWorkflow($workflowId)
+    {
+        $retObj = [];
+        $usedVariables = [];
+        $workflow = Workflow::find($workflowId);
+
+        if (!$workflow->is_verified || !$workflow->is_published) {
```

Comment:
Use middleware for this

Reply:
Authorisation middlewares don't work with guest so this is a workaround for this.